[Bug]: FieldState errors wrong typings

### Version Number

7.16.1

### Codesandbox/Expo snack

Latest version 7.16.1 does not seem to install on Codesandbox, see example repo instead

### Steps to reproduce

1. Clone example repo: https://github.com/StampixSMO/react-hook-form-fieldstate-errors-type
2. Install dependencies
3. Run `npm run compile`
4. Watch the TS error.

### Expected behaviour

The type of the errors object should not reflect the nested objects but only expose the `message` property.
This works fine in 7.15.4.

### Relevant log output

```
error TS2339: Property 'message' does
not exist on type 'DeepMap<DeepPartial<NestedValue<PhoneNumber>>, FieldError>'.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/src/__tests__/controller.test.tsx
+++ b/src/__tests__/controller.test.tsx
@@ -1254,6 +1254,6 @@ describe('Controller', () => {
 
     render(<App />);
 
-    expect(screen.getAllByRole('textbox').length).toEqual(3);
+    expect(screen.getAllByRole('textbox').length).toEqual(4);
   });
 });

--- a/src/__tests__/controller.test.tsx
+++ b/src/__tests__/controller.test.tsx
@@ -1239,12 +1239,7 @@ describe('Controller', () => {
             name="todos"
           />
           <Controller
-            render={({ field, fieldState }) => (
-              <>
-                <input {...field} />
-                <p>{fieldState.error?.message}</p>
-              </>
-            )}
+            render={({ fieldState }) => <p>{fieldState.error?.message}</p>}
             control={control}
             name="nestedValue"
           />

--- a/src/__tests__/controller.test.tsx
+++ b/src/__tests__/controller.test.tsx
@@ -1239,7 +1239,12 @@ describe('Controller', () => {
             name="todos"
           />
           <Controller
-            render={({ fieldState }) => <p>{fieldState.error?.message}</p>}
+            render={({ field, fieldState }) => (
+              <>
+                <input {...field} />
+                <p>{fieldState.error?.message}</p>
+              </>
+            )}
             control={control}
             name="nestedValue"
           />

--- a/src/__tests__/controller.test.tsx
+++ b/src/__tests__/controller.test.tsx
@@ -1241,7 +1241,7 @@ describe('Controller', () => {
           <Controller
             render={({ field, fieldState }) => (
               <>
-                <input {...field} />
+                <input {...{ ...field, value: field.value.test }} />
                 <p>{fieldState.error?.message}</p>
               </>
             )}

--- a/src/__tests__/controller.test.tsx
+++ b/src/__tests__/controller.test.tsx
@@ -8,7 +8,7 @@ import {
 } from '@testing-library/react';
 
 import { Controller } from '../controller';
-import { ControllerRenderProps } from '../types';
+import { ControllerRenderProps, NestedValue } from '../types';
 import { useFieldArray } from '../useFieldArray';
 import { useForm } from '../useForm';
 
@@ -1193,11 +1193,17 @@ describe('Controller', () => {
         test: string;
       };
       todos: string[];
+      nestedValue: NestedValue<{ test: string }>;
     };
 
     function App() {
       const { control } = useForm<FormValues>({
-        defaultValues: { firstName: '', deepNested: { test: '' }, todos: [] },
+        defaultValues: {
+          firstName: '',
+          deepNested: { test: '' },
+          todos: [],
+          nestedValue: { test: '' },
+        },
       });
 
       return (
@@ -1231,6 +1237,16 @@ describe('Controller', () => {
             )}
             control={control}
             name="todos"
+          />
+          <Controller
+            render={({ field, fieldState }) => (
+              <>
+                <input {...field} />
+                <p>{fieldState.error?.message}</p>
+              </>
+            )}
+            control={control}
+            name="nestedValue"
           />
         </form>
       );

--- a/src/types/controller.ts
+++ b/src/types/controller.ts
@@ -3,15 +3,22 @@ import * as React from 'react';
 import { RegisterOptions } from './validator';
 import {
   Control,
+  DeepMapImpl,
+  DeepPartial,
   FieldError,
-  FieldErrors,
   FieldPath,
   FieldPathValue,
   FieldValues,
   RefCallBack,
+  UnionLike,
   UnpackNestedValue,
   UseFormStateReturn,
 } from './';
+
+type ControllerFieldError<T> = DeepMapImpl<
+  DeepPartial<UnionLike<T>>,
+  FieldError
+>;
 
 export type ControllerFieldState<
   TFieldValues extends FieldValues = FieldValues,
@@ -20,9 +27,7 @@ export type ControllerFieldState<
   invalid: boolean;
   isTouched: boolean;
   isDirty: boolean;
-  error?: FieldErrors<FieldPathValue<TFieldValues, TFieldName>> extends any[]
-    ? FieldErrors<FieldPathValue<TFieldValues, TFieldName>>
-    : FieldError;
+  error?: ControllerFieldError<FieldPathValue<TFieldValues, TFieldName>>;
 };
 
 export type ControllerRenderProps<

--- a/src/types/controller.ts
+++ b/src/types/controller.ts
@@ -4,7 +4,7 @@ import { RegisterOptions } from './validator';
 import {
   Control,
   DeepMapImpl,
-  DeepPartial,
+  DeepPartialImpl,
   FieldError,
   FieldPath,
   FieldPathValue,
@@ -16,7 +16,7 @@ import {
 } from './';
 
 type ControllerFieldError<T> = DeepMapImpl<
-  DeepPartial<UnionLike<T>>,
+  DeepPartialImpl<UnionLike<T>>,
   FieldError
 >;
 

--- a/src/types/controller.ts
+++ b/src/types/controller.ts
@@ -3,22 +3,15 @@ import * as React from 'react';
 import { RegisterOptions } from './validator';
 import {
   Control,
-  DeepMapImpl,
-  DeepPartial,
   FieldError,
+  FieldErrors,
   FieldPath,
   FieldPathValue,
   FieldValues,
   RefCallBack,
-  UnionLike,
   UnpackNestedValue,
   UseFormStateReturn,
 } from './';
-
-type ControllerFieldError<T> = DeepMapImpl<
-  DeepPartial<UnionLike<T>>,
-  FieldError
->;
 
 export type ControllerFieldState<
   TFieldValues extends FieldValues = FieldValues,
@@ -27,7 +20,9 @@ export type ControllerFieldState<
   invalid: boolean;
   isTouched: boolean;
   isDirty: boolean;
-  error?: ControllerFieldError<FieldPathValue<TFieldValues, TFieldName>>;
+  error?: FieldErrors<FieldPathValue<TFieldValues, TFieldName>> extends any[]
+    ? FieldErrors<FieldPathValue<TFieldValues, TFieldName>>
+    : FieldError;
 };
 
 export type ControllerRenderProps<

--- a/src/types/utils.ts
+++ b/src/types/utils.ts
@@ -18,7 +18,7 @@ export type LiteralUnion<T extends U, U extends Primitive> =
   | T
   | (U & { _?: never });
 
-type DeepPartialImpl<T> = T extends NestedValue
+export type DeepPartialImpl<T> = T extends NestedValue
   ? T
   : T extends ReadonlyArray<any> | Record<any, unknown>
   ? DeepPartial<T>

